### PR TITLE
Add parallel front end goals

### PR DESCRIPTION
**Pull request by SparrowLii**
parallel front end goals from parallel rustc working group

cc @cjgillot @nnethercote @bjorn3 @Kobzol

### CODE REVIEW

File: src/2024h2/parallel-front-end.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The existing compiler testing framework is not sufficient for the parallel front end, and we need to enhance it to ensure the correct functionality of the parallel front end.
+
+The current parallel front end still has room for further improvement in compilation performance, such as parallelization of HIR lowering and macro expansion, and reduction of data contention under more threads (>= 16).
+
+We can use parallel front end in bootstrap to alleviate the problem of slow build of the whole project.
+
+### The next 6 months
+
+- Solve most of the existing issues (unless they are due to lack of rare hardware environment or are almost impossible to reproduce).
```
Comment (Kobzol) [inline src/2024h2/parallel-front-end.md:25]: These are good goals, but I wonder if they are perhaps a bit too ambitious for the next 6 months. Maybe we should first focus on better testing, bootstrap integration, and stabilization. And then later on more performance improvements?

**Review by Kobzol — COMMENTED** [review]
Looks great as a goal, thank you!

File: src/2024h2/parallel-front-end.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+| Metadata |                           |
+| -------- |---------------------------|
+| Owner(s) | [@SparrowLii]             |
+| Teams    | [Parallel Rustc WG] |
+| Status   | Existing problems need to be resolved to stabilize                          |
+
+## Motivation
+
+The parallel front end has been implemented in nightly, but there are still many problems that prevent it from being stable and used at scale.
```
Comment (nikomatsakis) [inline src/2024h2/parallel-front-end.md:11]: Sorry, I need to rework the tempalte to make this clearer, but can you add a summary that hits on "what you will do in next 6 months and what benefit to expect"?

Something like

> We will move rustc's support for parallel compilation closer to stability by resolving the XX known issues with rustc's parallel front-end and integrating parallel compilation into bootstrap. This fits into our larger goal of improving build times by N% by leveraging multiple cores.

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: src/2024h2/parallel-front-end.md (hunk 2 above)
Comment (nikomatsakis) [reply src/2024h2/parallel-front-end.md:11]: I updated the template etc -- see #31

Comment (SparrowLii) [reply src/2024h2/parallel-front-end.md:11]: Thanks! I gonna update this

File: src/2024h2/parallel-front-end.md (hunk 1 above)
Comment (lqd) [reply src/2024h2/parallel-front-end.md]: It also depends on many resources we expect to put into this goal, if we know who and how much people are expecting to be participating

Comment (SparrowLii) [reply src/2024h2/parallel-front-end.md]: Yea, In fact, the biggest problem is that many issues are difficult to reproduce. I think we should at least solve most of the ICE and deadlock issues

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**